components/validator/resources.rst

```diff
@@ -0,0 +1,200 @@
+.. index::
+    single: Validator; Loading Resources
+
+Loading Resources
+=================
+
+The Validator uses metadata to validate a value. This metadata defines how a
+class, array or any other value should be validated. When validating a class,
+each class contains its own specific metadata. When validating another value,
+the metadata must be passed to the validate methods.
+
+Class metadata should be defined somewhere in a configuration file, or in the
+class itself. The ``Validator`` needs to be able to retrieve this metadata
+from the file or class. To do that, it uses a set of loaders.
+
+.. seealso::
+
+    You'll learn how to define the metadata in :doc:`metadata`.
+
+The StaticMethodLoader
+----------------------
+
+The most basic loader is the
+:class:`Symfony\\Component\\Validator\\Mapping\\Loader\\StaticMethodLoader`.
+This loader will call a static method of the class in order to get the
+metadata for that class. The name of the method is configured using the
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addMethodMapping`
+method of the Validator builder::
+
+    use Symfony\Component\Validator\Validation;
+
+    $validator = Validation::createValidatorBuilder()
+        ->addMethodMapping('loadValidatorMetadata')
+        ->getValidator();
+
+Now, the retrieved ``Validator`` tries to find the ``loadValidatorMetadata()``
+method of the class to validate to load its metadata::
+
+    use Symfony\Component\Validator\Mapping\ClassMetadata;
+    use Symfony\Component\Validator\Constraints\NotBlank;
+    use Symfony\Component\Validator\Constraints\Length;
+
+    class User
+    {
+        protected $name;
+
+        public static function loadValidatorMatadata(ClassMetadata $metadata)
+        {
+            $metadata->addPropertyConstraint('name', new NotBlank());
+            $metadata->addPropertyConstraint('name', new Length(array(
+                'min' => 5,
+                'max' => 20,
+            )));
+        }
+    }
+
+.. tip::
+
+    You can call this method multiple times to add multiple supported method
+    names. You can also use
+    :method:`Symfony\\Component\\Validator\\ValidatorBuilder::addMethodMappings`
+    to set an array of supported method names.
+
+The FileLoaders
+---------------
+
+The component also provides 2 file loaders, one to load Yaml files and one to
+load XML files. Use 
+:method:`Symfony\\Component\\Validator\\ValidatorBuilder::addYamlMapping` or
```

Comment:
agree, I add it  (and we need also a note when we will talk about Form component integration)